### DISCUSSION TITLE
how i can control the properties/fields that can be retrieved from SharePoint inside my PnP core sdk

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [pnp/pnpcore#1398] [Q&A] [answered]
I have this pnp core sdk code:-

**var myList = await context.Web.Lists.GetByTitleAsync("Work Orders", p => p.Title,
                                                      p => p.Fields.QueryProperties(p => p.InternalName,
                                                                                    p => p.FieldTypeKind,
                                                                                    p => p.TypeAsString,
                                                                                    p => p.Title));
 int i = 0;
 var t = "";
 // Do a paged retrieval of the list items
 await foreach (var listItem in myList.Items)
 {
     // Do something with the list item
     t= listItem["Created"].ToString();
     i++;
 }**

now i am not loading the Created property, but how i were able to reference it inside my foreach loop? seems my code is loading all the properties, so how i can control which properties i want to get from SharePoint ? so i can have a better response time, where i can specify which properties need to be retrieved from SharePoint, instead of getting all properties ?

Thanks

**Comment by jansenbe** [comment, marked as answer]
@mvcsharepointdev : if you want to control the properties you get from SharePoint it's best to use a method that allows specifying a CAML query. These are options C and D specified in here: https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#reading-list-items
